src/App.jsx

```diff
@@ -14,33 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_REVERSE = 'reverse';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortedField) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortedField.includes(SORT_FIELD_ALPHABETICALLY)) {
+    preparedGoods.sort((good1, good2) => {
+      return good1.localeCompare(good2);
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortedField.includes(SORT_FIELD_LENGTH)) {
+    preparedGoods.sort((good1, good2) => {
+      return good1.length - good2.length;
+    });
+  }
+
+  if (sortedField.includes(SORT_FIELD_REVERSE)) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState([]);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField);
+
+  function updateSortField(field) {
+    if (field === SORT_FIELD_REVERSE) {
+      setSortField(
+        sortField.includes(field)
+          ? [...sortField].filter(currentField => currentField !== field)
+          : [...sortField, field],
+      );
+    } else {
+      setSortField(
+        field === SORT_FIELD_LENGTH
+          ? [...sortField, field].filter(
+              currentField => currentField !== SORT_FIELD_ALPHABETICALLY,
+            )
+          : [...sortField, field].filter(
+              currentField => currentField !== SORT_FIELD_LENGTH,
+            ),
+      );
+    }
+  }
```

Comment:
The idea to keep all sort fields in (btw - the better naming would be `flags`) array is good, but how you switch them is over-complicated. In fact, it makes sense to treat them in more generic way